### DISCUSSION TITLE
How to Estimate Gas for createListing

### DISCUSSION THREAD

**Discussion by ekhishigee** [ProjectOpenSea/opensea-js#1378] [Q&A]
I would like to learn how to estimate gas for creating listings. This knowledge is essential for preventing insufficient gas errors when the wallet balance is insufficient for creating listings.

**Comment by ryanio** [comment]
Create listing does not use any gas, as it is an off chain action (unless you have to approve the contract first). You can check if the connected wallet has the sufficient funds with provider.getBalance for native token and balanceOf for ERC20s

**Reply by ekhishigee** [reply]
@ryanio 
Hi, 
It's possible that the action triggered a 'Set Approval For All' transaction, which can be verified on etherscan https://sepolia.etherscan.io/tx/0x61b5a20601c5e264b7e908762df198d9c5910d2e6464ff7ebdb0bc7c4eccdebc. When attempting to call the 'create listing' method of the OpenSea SDK with a wallet balance of 0, the transaction failed with an 'insufficient funds' error. This suggests that the transaction may require additional funds due to the 'Set Approval For All' action.

**Reply by ekhishigee** [reply]
@ryanio 
Just circling back on my previous message regarding it. I wanted to see if you had a chance to review it and if there are any updates or next steps we should discuss.

**Reply by ryanio** [reply]
If you want to estimate the gas for the approval transaction you need to use the underlying `@opensea/seaport-js` library to get the transaction details before sending and you can call estimateGas() on it. The gas will usually be the same it's moreso the gas price which fluctuates.
